library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerDownloader.java

```diff
@@ -54,28 +54,40 @@
         this.callbackThrottleCreator = callbackThrottleCreator;
     }
 
-    public void download(Batch batch, Map<DownloadBatchId, DownloadBatch> downloadBatchMap) {
-        DownloadBatch downloadBatch = DownloadBatchFactory.newInstance(
+    void creteDownloadBatchAndDownload(Batch batch, Map<DownloadBatchId, DownloadBatch> downloadBatchMap) {
+        executor.submit(() -> {
+            DownloadBatch downloadBatch = DownloadBatchFactory.newInstance(
                 batch,
                 fileOperations,
                 downloadsBatchPersistence,
                 downloadsFilePersistence,
                 callbackThrottleCreator.create(),
                 connectionChecker
-        );
+            );
 
-        downloadBatchMap.put(downloadBatch.getId(), downloadBatch);
-        download(downloadBatch, downloadBatchMap);
+            downloadBatchMap.put(downloadBatch.getId(), downloadBatch);
+            downloadSync(downloadBatch, downloadBatchMap);
+        });
     }
 
-    public void download(DownloadBatch downloadBatch, Map<DownloadBatchId, DownloadBatch> downloadBatchMap) {
+    private void downloadSync(DownloadBatch downloadBatch, Map<DownloadBatchId, DownloadBatch> downloadBatchMap) {
```

Comment:
Not sure I agree with this name. I would avoid naming something as synchronous unless it is exposed publicly and clients of the application need to handle the threading themselves. We know when something is synchronous or not by whether we have added it to an executor and whether it is annotated with the `@WorkerThread`

Reply:
I agree, and I added the `@WorkerThread` annotation. However, I'd need another name for to distinguish this and the `download()` method submitting to the executor

Reply:
`submitDownload`?

For the sync naming, they do this a lot if nodejs land where there's a 
```
readFile(callback)
readFileSync()
```